src/main/java/com/astrapay/qris/cpm/validation/HexLengthValidator.java

```diff
@@ -0,0 +1,42 @@
+package com.astrapay.qris.cpm.validation;
+
+import com.astrapay.qris.QrisHexConverter;
+import com.astrapay.qris.cpm.QrisCpmEncoder;
+import com.astrapay.qris.cpm.validation.constraints.HexLength;
+import com.astrapay.qris.mpm.object.QrisDataObject;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.Map;
+
+public class HexLengthValidator implements ConstraintValidator<HexLength, Map<String, QrisDataObject>> {
+
+    private String id;
+    private int min;
+    private int max;
+
+    @Autowired
+    QrisCpmEncoder qrisCpmEncoder;
+
+    @Autowired
+    QrisHexConverter qrisHexConverter;
+
+    @Override
+    public void initialize(HexLength constraintAnnotation) {
+        this.id = constraintAnnotation.id();
+        this.min = constraintAnnotation.min();
+        this.max = constraintAnnotation.max();
+    }
+
+    @Override
+    public boolean isValid(Map<String, QrisDataObject> value, ConstraintValidatorContext context) {
```

Comment:
Add context to this method especially for what happens in line 38 for in the current version.